Add docker-compose.yml and corresponding .env

WolframRavenwolf opened the pull request:
Noticed vLLM is still missing a Composefile, so here's one. You can set the most important options (like which model to use) in the corresponding .env file, or by editing the command lines in docker-compose.yml. Will work out of the box, but is also a solid base to build upon.

Commit:
Add docker-compose.yml and corresponding .env

WolframRavenwolf commented:
Could anyone please review and merge? Thanks!

DarkLight1337 commented:
@dtrifiro do you have time to consider this?

dtrifiro commented on .env:
```diff
@@ -0,0 +1,8 @@
+#PORT=8000
```
Can we get rid of this `.env` file and just provide an example dockerfile with values that can be edited?

WolframRavenwolf replied:
.env file is optional, but good practice - users should edit that instead of the Compose file directly

dtrifiro commented on docker-compose.yml:
```diff
@@ -0,0 +1,33 @@
+services:
+  vllm:
+    command: # python3 -m vllm.entrypoints.openai.api_server
+      ##########################################################################
```
```suggestion
```

WolframRavenwolf replied:
Cleaned up Docker Compose file by removing redundant comments

dtrifiro commented on docker-compose.yml:
```diff
@@ -0,0 +1,33 @@
+    volumes:
+      #- ~/.cache/huggingface:/root/.cache/huggingface
+      - ${HOME:?err}/.cache/huggingface/hub:/root/.cache/huggingface/hub
+      - /etc/localtime:/etc/localtime:ro
```
Are these two volume mounts for time/timezone actually required?

WolframRavenwolf replied:
Replaced localtime/timezone mount points with TZ environment variable so users can set the container to their local timezone

dtrifiro commented:
Looks good overall, can we add some documentation in `README.md` for a quick start using docker compose though? Thanks!

WolframRavenwolf committed:
Merge branch 'vllm-project:main' into docker-compose

WolframRavenwolf committed:
Adjust docker-compose.yml and corresponding .env

WolframRavenwolf commented:
Adjusted docker-compose.yml and corresponding .env:

- .env file is optional, but good practice - users should edit that instead of the Compose file directly
- Cleaned up Docker Compose file by removing redundant comments
- Removed deprecated Compose version information
- Switched default model to nvidia/Mistral-NeMo-Minitron-8B-Instruct (mistralai/Mistral-7B-Instruct-v0.2 is outdated and their models are gated, and facebook/opt-125m isn't very relevant)
- If HOME environment variable is not set, cache is created in Composefile's directory instead of throwing an error
- Replaced localtime/timezone mount points with TZ environment variable so users can set the container to their local timezone

Please confirm if this looks good to you, then I'll look into writing some documentation in the README.md.

hmellor commented:
This is probably now superseded by https://github.com/hmellor/production-stack
